benchmarks/graphics_pipeline/main.cpp

```diff
@@ -844,28 +844,54 @@ void ProjApp::Render()
                     frame.cmd->DrawIndexed(indexCount, /* instanceCount = */ 1, firstIndex);
                 }
             }
+        }
+        frame.cmd->EndRenderPass();
 
-            // Noise Quads
-            if (pNoiseQuadsCount->GetValue() > 0) {
-                // Draw noise quads
-                frame.cmd->BindGraphicsPipeline(mNoiseQuads.pipeline);
-                frame.cmd->BindVertexBuffers(1, &mNoiseQuads.vertexBuffer, &mNoiseQuads.vertexBinding.GetStride());
+        // =====================================================================
+        // Fullscreen quads renderpasses
+        // =====================================================================
+        if (pNoiseQuadsCount->GetValue() > 0) {
+            for (size_t i = 0; i < pNoiseQuadsCount->GetValue(); ++i) {
+                currentRenderPass = swapchain->GetRenderPass(imageIndex);
+                PPX_ASSERT_MSG(!currentRenderPass.IsNull(), "render pass object is null");
+
+                frame.cmd->BeginRenderPass(currentRenderPass);
+                {
+                    // Draw noise quads
+                    frame.cmd->BindGraphicsPipeline(mNoiseQuads.pipeline);
+                    frame.cmd->BindVertexBuffers(1, &mNoiseQuads.vertexBuffer, &mNoiseQuads.vertexBinding.GetStride());
 
-                for (size_t i = 0; i < pNoiseQuadsCount->GetValue(); ++i) {
                     uint32_t noiseQuadRandomSeed = (uint32_t)i;
                     frame.cmd->PushGraphicsConstants(mNoiseQuads.pipelineInterface, 1, &noiseQuadRandomSeed);
                     frame.cmd->Draw(4, 1, 0, 0);
                 }
-            }
+                frame.cmd->EndRenderPass();
 
-            DrawImGui(frame.cmd);
+                // Force resolve by transitioning image layout
+                frame.cmd->TransitionImageLayout(currentRenderPass->GetRenderTargetImage(0), PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_RENDER_TARGET, grfx::RESOURCE_STATE_SHADER_RESOURCE);
+                frame.cmd->TransitionImageLayout(currentRenderPass->GetRenderTargetImage(0), PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_SHADER_RESOURCE, grfx::RESOURCE_STATE_RENDER_TARGET);
```

Comment:
Not sure if transition back to  RENDER_TARGET should be included in the timestamp range, since the first one should resolve the image already.

Reply:
Hmm, are you suggesting we should add more timestamps and do some addition/subtraction to specifically exclude all the noise quads transitions from `shader_resource -> render_target`?

Reply:
I'm not suggest anything, the changes seems fine.
I'm not sure if there's any additional cost associated with the second transition, and would like someone more knowledgeable to clarify.